### PR TITLE
Adding ability to listen to multiple kafka10 topics via comma separated list

### PR DESCRIPTION
**Pull request by danielkwinsor**
Useful for cases like span info from different environments coming from different topics, etc.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
seems handy, but would need a test and updates on the config docs. Also, we should make sure that this is something others agree with as I don't know the support implications of always using ConsumerRebalanceListener.

Sorry we don't have a kafka mailer at the moment. pinging @dgrabows as he touched this recently

**Comment by dgrabows** [comment]
This change make sense to me.

I don't think the ConsumerRebalanceListener is a concern. That was already there and not introduced with this change. It gets used for the worker to maintain a list of currently assigned partitions. That list is only used in the tests for multiple consumer threads, where the test needs to wait until each consumer has some assignment before proceeding.

The listener will be invoked each time partitions are reassigned in the consumer group, which will only happen once per restart if running a single instance of the collector. If running multiple collectors, invocations will occur whenever a collector is started, stopped, or looses connectivity to the kafka brokers. Those should be relatively infrequent events.

**Comment by codefromthecrypt** [comment]
ok great. @danielkwinsor mind adding a test? (can be as simple as checking that split worked)

lemme know if you need help with this

**Comment by danielkwinsor** [comment]
I was figuring a request for tests would come :)  I definitely will do it when I can fit it between my heavy work queue.  My only concern is that area didn't look easy to mock, so I don't know when I'll be able to get to it.

Manual testing did show this worked out with single or multiple topics, but I didn't test things like whitespace, etc.

**Comment by codefromthecrypt** [comment]
looks like you might be able to write a simple test that uses package access to check kafkaConsumer.subscription()

basically read-back that to show a simple split works. wanna try?

**Comment by codefromthecrypt** [comment]
k very least this needs an update to the docs KafkaCollector.Builder and in zipkin-server/README like..

```
    * `KAFKA_TOPIC`: Topic zipkin spans will be consumed from. Defaults to "zipkin"
      * When Kafka 0.10 is in use, multiple topics may be specified if comma delimited
```

**Comment by codefromthecrypt** [comment]
All good or did I scare you off? :)

**Comment by danielkwinsor** [comment]
All good... in the process of adding docs and tests, but rebased from
upstream and it wiped out my commit, then auto closed.  Expect by EoD.

**Comment by danielkwinsor** [comment]
I spent too long on the unit test and gave up.  It's much too hard to mock something coming from a constructor without powermockito.  Here's the code plus docs.

File: zipkin-autoconfigure/collector-kafka10/README.md [hunk 1]
```diff
@@ -107,14 +107,14 @@ $ KAFKA_BOOTSTRAP_SERVERS=broker1:9092.local,broker2.local:9092 \
...
-Alternate topic name:
+Alternate topic name(s):
 
 ```bash
 $ KAFKA_BOOTSTRAP_SERVERS=127.0.0.1:9092 \
     java \
     -Dloader.path='zipkin-autoconfigure-collector-kafka10-module.jar,zipkin-autoconfigure-collector-kafka10-module.jar!/lib' \
     -Dspring.profiles.active=kafka \
-    -Dzipkin.collector.kafka.topic=zapkin \
+    -Dzipkin.collector.kafka.topic=zapkin,zipken \
```
Comment (codefromthecrypt) [inline zipkin-autoconfigure/collector-kafka10/README.md:117]: hehe

**Comment by codefromthecrypt** [comment]
Thanks very much @danielkwinsor! (and thanks for the timeboxed attempt on tests.. I'll copy you on a test commit)

**Pull request merged**